### PR TITLE
Data loader elasticsearch

### PR DESCRIPTION
**Pull request by walterra**
As discussed on your Community Slack, I'd like to contribute a data loader example for Elasticsearch.

The original repo I created this in can be found here:
https://github.com/walterra/observable-framework-data-loader-elasticsearch

A live view can be found here: 
https://walterra.observablehq.cloud/framework-example-loader-elasticsearch/

This Observable Framework example demonstrates how to write a TypeScript data loader that runs a query on Elasticsearch using the [Elasticsearch Node.js client](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/index.html). The data loader lives in [`src/data/kibana_sample_data_logs.csv.ts`](./src/data/kibana_sample_data_logs.csv.ts) and uses the helper [`src/data/es_client.ts`](./src/data/es_client.ts).

Similar to #1477, I added the output of the data loader as a static file too.

To fully reproduce the example, you need to have a setup with both Elasticsearch and Kibana running to create the sample data. Here's how to set up both on macOS:

```bash
# Download and run Elasticsearch
curl -O https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-8.14.1-darwin-x86_64.tar.gz
gunzip -c elasticsearch-8.14.1-darwin-x86_64.tar.gz | tar xopf -
cd elasticsearch-8.14.1
./bin/elasticsearch

# Next, in another terminal tab, download and run Kibana
curl -O https://artifacts.elastic.co/downloads/kibana/kibana-8.14.1-darwin-x86_64.tar.gz
gunzip -c kibana-8.14.1-darwin-x86_64.tar.gz | tar xopf -
cd kibana-8.14.1
./bin/kibana
```

The commands for both will output instructions how to finish the setup with security enabled. Once you have both running, you can create the sample data in Kibana via this URL: http://localhost:5601/app/home#/tutorial_directory/sampleData

Finally, create the `.env` file with the credentials shared for the user `elastic` that were logged when starting Elasticsearch like this. To get the CA fingerprint for the config, run the following command from the directory you started installing Elasticsearch:

```
openssl x509 -fingerprint -sha256 -noout -in ./elasticsearch-8.14.1/config/certs/http_ca.crt
```

```
ES_NODE="https://elastic:<PASSWORD>@localhost:9200"
ES_CA_FINGERPRINT="<CA_FINGERPRINT>"
ES_UNSAFE_TLS_REJECT_UNAUTHORIZED="FALSE"
```

### CODE REVIEW

File: examples/loader-elasticsearch/src/index.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+For the data loader to authenticate with your Elasticsearch cluster, you need to set the environment variables defined in the helper. If you use GitHub, you can use [secrets in GitHub Actions](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions) to set environment variables; other platforms provide similar functionality for continuous deployment. For local development, we use the `dotenv` package, which allows environment variables to be defined in a `.env` file which lives in the project root and looks like this:
+
+```
+ES_NODE="https://USERNAME:PASSWORD@HOST:9200"
+```
+
+<div class="warning">
+The `.env` file should not be committed to your source code repository; keep your credentials secret.
+</div>
```
Comment (Fil) [inline examples/loader-elasticsearch/src/index.md:156]:
```suggestion
<div class="warning">

The `.env` file should not be committed to your source code repository; keep your credentials secret.

</div>
```


adding empty lines should allow framework to convert the markdown (in this case, the backticks around .env)

File: examples/loader-elasticsearch/src/index.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+          rejectUnauthorized: false,
+        },
+      }
+    : {}),
+});
+```
+
+<div class="note">
+To run this data loader, you’ll need to install `@elastic/elasticsearch`, `d3-dsv` and `dotenv` using your preferred package manager such as npm or Yarn.
+</div>
```
Comment (Fil) [inline examples/loader-elasticsearch/src/index.md:146]:
```suggestion
<div class="note">

To run this data loader, you’ll need to install `@elastic/elasticsearch`, `d3-dsv` and `dotenv` using your preferred package manager such as npm or Yarn.

</div>
```

File: examples/loader-elasticsearch/src/index.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+const logs = FileAttachment("./data/kibana_sample_data_logs.csv").csv({typed: true});
+```
+
+The `logs` table has three columns: `date`, `count` and `response_code`. We can display the table using `Inputs.table`.
+
+```js echo
+Inputs.table(logs)
+```
+
+Lastly, we can pass the table to `Plot.plot` to make a line chart.
```
Comment (Fil) [inline examples/loader-elasticsearch/src/index.md:170]:
```suggestion
Lastly, we can pass the table to Observable Plot to make a line chart.
```

**Comment by walterra** [comment]
Thanks for the initial feedback, I updated `index.md` with the suggestions. Also added a link to [examples/README.md](https://github.com/observablehq/framework/pull/1490/commits/aebd9108ecac1a97ab55f5588ffb04bb0f55d069).

**Comment by Fil** [comment]
Thanks for the very detailed procedure — I was able to test it locally.

I now have just a few more suggestions:

* we should use framework version `1.9.0`, or `latest`, in `package.json`.
* Regarding the sample data sets, indicate that we need dataset "Sample web logs"; they're called `kibana_sample_data_logs` in the loader).
* I personally added double quotes around the values in `.env`, but it's not mandatory.
* `ES_CA_FINGERPRINT` was already shown in the logs of elasticsearch, so the manual ssl step seemed almost redundant. The only difference is that in one case it is written in lowercase and without colons, when the correct format has uppercase and colons (`9191e5abbecb472806b453…` vs. `91:91:E5:AB:BE:CB:47:28:06:B4:53:…`).

Another note in passing (but probably not actionable): as I was testing the setup, I was blocked as the enrollment string linked to IP address 10.100.0.2. But this address would not connect, and this prevented the installation from working at all. I realized I had to disable NordVPN, delete the whole es installation, and start again from scratch… and it worked. (I guess it's more an issue with NordVPN than with ES, but figured maybe you'd want to investigate this.)

**Comment by Fil** [comment]
> Did the lowercase version without colons work for you?

Nope :). To clarify, it would have been great to have the correct format in that place and not have to do the call to `openssl`, as this is always stressful—but it's not a blocker for this PR!

File: examples/loader-elasticsearch/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+[Framework examples →](../)
+
+# Elasticsearch data loader
+
+View live: <https://observablehq.observablehq.cloud/framework-example-loader-elasticsearch/>
+
+This Observable Framework example demonstrates how to write a TypeScript data loader that runs a query on Elasticsearch using the [Elasticsearch Node.js client](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/index.html). The data loader lives in [`src/data/kibana_sample_data_logs.csv.ts`](./src/data/kibana_sample_data_logs.csv.ts) and uses the helper [`src/data/es_client.ts`](./src/data/es_client.ts).
+
+To fully reproduce the example, you need to have a setup with both Elasticsearch and Kibana running to create the sample data. Here's how to set up both on macOS:
```
Comment (Fil) [inline examples/loader-elasticsearch/README.md:9]:
```suggestion
To fully reproduce the example, you need to have a setup with both Elasticsearch and Kibana running to create the sample data. Here’s how to set up both on macOS:
```

File: examples/loader-elasticsearch/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+./bin/elasticsearch
+
+# Next, in another terminal tab, download and run Kibana
+curl -O https://artifacts.elastic.co/downloads/kibana/kibana-8.14.1-darwin-x86_64.tar.gz
+gunzip -c kibana-8.14.1-darwin-x86_64.tar.gz | tar xopf -
+cd kibana-8.14.1
+./bin/kibana
+```
+
+The commands for both will output instructions how to finish the setup with security enabled. Once you have both running, you can create the "Sample web logs" dataset in Kibana via this URL: http://localhost:5601/app/home#/tutorial_directory/sampleData
```
Comment (Fil) [inline examples/loader-elasticsearch/README.md:25]:
```suggestion
The commands for both will output instructions how to finish the setup with security enabled. Once you have both running, you can create the “Sample web logs” dataset in Kibana via this URL: http://localhost:5601/app/home#/tutorial_directory/sampleData
```

**Review by Fil — COMMENTED** [review]
Add curly quotes before we merge (I don't think I am allowed to push to this branch?).

I've deployed the project to https://observablehq.observablehq.cloud/framework-example-loader-elasticsearch/

**Comment by Fil** [comment]
hmm no, that was my oversight; thanks for reminding me!

To deploy the examples we have to include the common configuration file which adds the specific headers:

```
yarn deploy -c ../observablehq.config.js -c ./observablehq.config.js
```

**Review by Fil — APPROVED** [review]
thank you!

**Pull request merged**